packages/create-plugin/templates/backend-app/pkg/plugin/app_test.go

```diff
@@ -0,0 +1,135 @@
+package plugin
+
+import (
+	"bytes"
+	"context"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"net/http"
+	"testing"
+)
+
+// mockCallResourceResponseSender implements backend.CallResourceResponseSender
+// for use in tests.
+type mockCallResourceResponseSender struct {
+	response *backend.CallResourceResponse
+}
+
+// Send sets the received *backend.CallResourceResponse to s.response
+func (s *mockCallResourceResponseSender) Send(response *backend.CallResourceResponse) error {
+	s.response = response
+	return nil
+}
+
+func TestCallResource(t *testing.T) {
+	// Initialize app
+	inst, err := NewApp(backend.AppInstanceSettings{})
```

Comment:
We can do that, however I did it this way because `httptest` works with `http.Response` and `http.ResponseRecorder` (which implements `io.Writer`), so it works with stdlib `http.HandlerFunc`.

This works well to test the actual handlers logic, however I wanted to test that `CallResource` calls the correct handler and returns the correct response (more end to end rather than unit testing the actual handlers).

I did that because the handlers return just dummy data, however for a "real" app I agree it makes more sense to test the handlers alone with `httptest` and then maybe have an additional test that involves `CallResource`.

I probably went a bit overkill on the tests anyways since this is just a template. Maybe it's a better idea to simplify this and just have a simple test for the handler's logic with `httptest`, and another one with `CallResource` to demonstrate that the `httpadapter` works. After all, those tests are going to be deleted by the plugin author anyways after they implement their own resources.

What do you think?

Reply:
Gotcha. That make sense :+1: 

> Maybe it's a better idea to simplify this and just have a simple test for the handler's logic with `httptest`, and another one with `CallResource` to demonstrate that the `httpadapter` works. After all, those tests are going to be deleted by the plugin author anyways after they implement their own resources.

Yeah that probably makes more sense and could make the test code more readable/understandable. Even if plugin author implements their own resources, including some boilerplate test code is probably very helpful and can be changed/used as template to write tests for their own resources.

Reply:
I have updated the tests so that there's one for the handlers and another one that involves CallResource. Is it better now?